diesel_cli/src/main.rs

```diff
@@ -282,6 +283,14 @@ fn regenerate_schema_if_file_specified(matches: &ArgMatches) -> Result<(), crate
                     .map_err(|e| crate::errors::Error::IoError(e, Some(path.to_owned())))?;
 
                 if schema.lines().ne(old_buf.lines()) {
+                    // it's fine to leak here, we will
+                    // exit the application anyway soon
+                    let label = path.file_name().expect("We have a file name here");
+                    let label = label.to_string_lossy().into_owned().leak();
```

Comment:
The `leak()` here is probably not necessary, but as the comment says it's fine.

Reply:
https://docs.rs/similar-asserts/latest/similar_asserts/struct.SimpleDiff.html#method.from_str unfortunately requires that the labels are `&'static str`, so yes that `leak()` is strictly speaking not required if we would use a static label instead.

Reply:
Ah, d'oh, I missed that the labels had a different lifetime from the values. This has actually been relaxed in main of similar-asserts but not yet released. I've added a comment to the relevant issue: https://github.com/mitsuhiko/similar-asserts/pull/9#issuecomment-2327210092